[Bug] 32-bit platforms using gcc need `libatomic` linking: `Undefined symbols for architecture ppc: "___atomic_load_8"`

### Describe the issue

```
[100%] Linking CXX executable CorsixTH.app/Contents/MacOS/CorsixTH
cd /opt/local/var/macports/build/corsixth-18d2ae2e/work/build/CorsixTH && /opt/local/bin/cmake -E cmake_link_script CMakeFiles/CorsixTH.dir/link.txt --verbose=ON
Undefined symbols for architecture ppc:
  "___atomic_load_8", referenced from:
      midi_player::init_playback(playback_state&) in libCorsixTH_lib.a(midi_player.cpp.o)
      __ZN11midi_player13playback_loopEO14playback_state in libCorsixTH_lib.a(midi_player.cpp.o)
  "___atomic_store_8", referenced from:
      midi_player::set_volume(double) in libCorsixTH_lib.a(midi_player.cpp.o)
ld: symbol(s) not found for architecture ppc
collect2: error: ld returned 1 exit status
```

### Steps to Reproduce

1. Run the build on ppc, arm or mips.

### Expected Behaviour

Configure stage should test for 8-byte atomics support.

### Have you enabled any current CorsixTH Experimental Features?

- [ ] Right mouse panning
- [ ] Falling actions (applies to CorsixTH versions < 0.70.0)

### Save Game

Save game attached to report: Yes/No

### CorsixTH Version

0.70.0

### Operating System

macOS 10.6.8

### Theme Hospital Version

CD

### Gamelog.txt

```yaml

```

### Additional Information

_No response_

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,6 +72,10 @@ if(ENABLE_UNIT_TESTS)
   enable_testing()
 endif()
 
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
 if(MINGW)
   set(CMAKE_EXE_LINKER_FLAGS "-static-libgcc -static-libstdc++")
 endif()

--- a/CorsixTH/CMakeLists.txt
+++ b/CorsixTH/CMakeLists.txt
@@ -18,6 +18,15 @@ if(MSVC)
   add_definitions(-D_CRT_SECURE_NO_WARNINGS)
 endif()
 
+include(CheckCXXSourceCompiles)
+check_cxx_source_compiles("
+#include <atomic>
+int main() {
+  std::atomic<double> d {0.0};
+  return static_cast<int>(d.load());
+}
+" ATOMIC_BUILTIN)
+
 set(CORSIX_TH_SEARCH_LOCAL_DATADIRS ${SEARCH_LOCAL_DATADIRS})
 
 set(CORSIX_TH_INTERPRETER_NAME CorsixTH.lua)
@@ -91,6 +100,10 @@ target_link_libraries(CorsixTH PRIVATE CorsixTH_lib)
 
 if(UNIX AND NOT APPLE)
   set_target_properties(CorsixTH PROPERTIES OUTPUT_NAME corsix-th)
+endif()
+
+if(NOT ATOMIC_BUILTIN)
+  target_link_libraries(CorsixTH_lib PRIVATE atomic)
 endif()
 
 # Generate source files list

